app/src/main/java/fr/free/nrw/commons/nearby/NearbyActivity.java

```diff
@@ -262,6 +262,7 @@ protected void onResume() {
         super.onResume();
         lockNearbyView = false;
         checkGps();
+        setupPlaceList(this);
```

Comment:
instead of calling `setupPlaceList(this)` you can use `refreshView` method. It has appropriate checks in place to determine if the list needs to be refreshed or not

Reply:
But won't `refreshView` instantly return if `lockNearbyView` is true?

Reply:
Edit: Whoops, I should have re-read that. Sorry lol. Will try it!